erpnext/accounts/doctype/subscription/subscription.py

```diff
@@ -371,10 +380,12 @@ def create_invoice(self, prorate):
 
 		# Subscription is better suited for service items. I won't update `update_stock`
 		# for that reason
-		items_list = self.get_items_from_plans(self.plans, prorate)
+		items_list, temp_currency = self.get_items_from_plans(self.plans, prorate)
 		for item in items_list:
 			item['cost_center'] = self.cost_center
 			invoice.append('items', item)
+			# currency
+			invoice.currency = temp_currency
```

Comment:
@MohammedAlNazili this doesn't look like a correct fix to me. One subscription could have multiple subscription plans and by this logic, the invoice will be made in the currency of the last subscription plan in the child table.
The invoice current should be the party account currency and independent of the subscription plan

I have tried to resolve this in the following PR. Please update your site and check if this helps
https://github.com/frappe/erpnext/pull/28916

For now closing this PR

Reply:
I tried to make subscriptions accept subscription plans in one currency in entry and by accepting one currency I can take first or last.
but with Customer currency, I think it is better.
Thanks